lib/connection.js

```diff
@@ -1,4 +1,5 @@
-const { dbconn } = require('idb-connector');
+const { dbconn, SQL_ATTR_DBC_SYS_NAMING, SQL_TRUE } = require('idb-connector');
+const Statement = require('./statement');
```

Comment:
I understand why you moved this up but the `const Statement = require('./statement');` was intentionally placed within getStatement() to guard against circular dependency issue.

Currently there is a way to Create a Statement from a Connection.

Connection -> Statement

There is also an unorthodox way of creating a Connection implicitly when one is not provided to the [Statement constructor](https://github.com/IBM/nodejs-idb-pconnector/blob/master/lib/statement.js#L11).

Statement -> Connection

This causes a [circular dependency issue](https://nodejs.org/api/modules.html#modules_cycles).

Running the tests after cloning your branch shows that a bunch of the Statement tests fail.

```
  4) Statement Class Tests
       bindParams
         associate parameter markers in an SQL to app variables:
     TypeError: Connection is not a constructor
      at new Statement (lib/statement.js:11:28)
      at Context.it (test/statementTest.js:90:25)


```

Reply:
Agreed. If we want to improve how this circular dependency is solved (if that's possible), then we should do it in a separate branch. @worksofliam can you clean up this PR to have `setLibraryList` call `getStatement` instead?

Reply:
After PR https://github.com/IBM/nodejs-idb-pconnector/pull/43, statement.js now has a createConnection() which hold the require of connection.